### DISCUSSION TITLE
What Front End to use for the Tauri App

### DISCUSSION THREAD

**Discussion by gmpinder** [blue-build/.github#13] [General]
So I've seen @xynydev getting interested in having a [Tauri](https://tauri.app/) app. So I'm opening this discussion to figure out how we want to setup the front end framework. The Tauri CLI gives us several options on how to setup an app. Here are the options:

```
? Choose which language to use for your frontend ›
❯ Rust  (cargo)
❯ TypeScript / JavaScript  (pnpm, yarn, npm, bun)

? Choose your package manager ›
❯ pnpm
❯  yarn
❯ npm
❯  bun

? Choose your UI template ›
❯ Vanilla
❯ Vue
❯ Svelte
❯ React
❯ Solid
❯ Angular
❯ Preact

? Choose your UI flavor ›
❯ TypeScript
❯ JavaScript
```

**Comment by gmpinder** [comment]
My personal choice would be TS, NPM, and Svelte.

TS - Type safety helps keep complex apps in control
NPM - The most basic for packaging (though I wouldn't mind trying bun)
Svelte - Because that seems to be @xynydev 's favorite framework and would like their help in this project

**Reply by xynydev** [reply]
I'd go with `ts`, `pnpm`, and `svelte`. TS for type safety, pnpm is really fast and 100% npm compatible, and Svelte has just really nice DX.

**Comment by xynydev** [comment]
I can handle the UI and design side on this project, I already have plans for which building blocks to use.

I can also handle browser-side logic.

I would want to support both a web version and a local version. For this we'd need a unified TS API for doing the basic repo file editing steps, both for the filesystem and and the GitHub file editing API.

**Comment by gmpinder** [comment]
What are folks thought on the license for this app? I think this would be a good opportunity for a Copyleft license like GPLv3

**Reply by xynydev** [reply]
If we go with Copyleft, I'd go AGPL, as that has additional protections for running as a web app. Actually yeah, that'd be great. Because our tooling can be freely used by anyone, but an user interface could have the potential to be turned into a SAAS by a third-party, and we'd want to keep that open.

**Comment by xynydev** [comment]
*Just using this thread to talk about this project more generally for contingency.*

I saw Fireship talk about [Google Cloud Run](https://cloud.google.com/run/) and we might be technically able to automate the cosign key generation in a remote container for free, to make the setup a bit easier. Just as long as requests don't exceed 2mil per month. Or we don't find a way to do this with Cloudflare Functions or something else easier.

That approach, if it's developed, should be optional, as it might raise security concerns in people. Could mitigate some of that by generating a salt clientside, the encrypting the keys with that, before passing them through HTTPS.

**Comment by xynydev** [comment]
I think we should figure out a name soon, and create a repo so that stuff can be discussed there.

**Comment by xynydev** [comment]
Created a repo with AGPL+Tauri+Svelte and all the goodies. Focusing on some scaffolding first, then MPV with re-implementing the current features of images-website in a saner way, and after that we can focus on more big picture stuff. https://github.com/blue-build/workshop
